capreolus/task/rerank.py

```diff
@@ -34,13 +39,36 @@ def describe(config, modules):
 
 
 def train(config, modules):
-    output_path = _pipeline_path(config, modules)
-    print("**** got train")
+    searcher = modules["searcher"]
+    benchmark = modules["benchmark"]
+    reranker = modules["reranker"]
+    extractor = reranker.modules["extractor"]
+
+    topics_fn = benchmark.topic_file
+    metric = "map"
+    searcher.index.create_index()
+    search_results_folder = searcher.query_from_file(topics_fn, os.path.join(searcher.get_cache_path(), benchmark.name))
+    best_search_run_path = evaluator.search_best_run(search_results_folder, benchmark, metric)["path"]
+    best_search_run = searcher.load_trec_run(best_search_run_path)
+    train_dataset = TrainDataset(best_search_run, benchmark, extractor)
+    train_dataloader = torch.utils.data.DataLoader(train_dataset, batch_size=config["batch"])
+    # Call the reranker's score() method here
```

Comment:
We will pass the train_dataloader and the reranker instance to the trainer module here I guess